### ISSUE TITLE
Plugin Causes Failed Netlify Deploy Due To Vast Memory Use

### ISSUE THREAD

**Issue by Undistraction** [Creatiwity/gatsby-plugin-favicon#40]
Using this plugin when deploying to Netlify results in the build failing due to the plugin using a huge amount of memory.

In Netlify's deploy logs you will see:

```
11:41:18 AM: success Building production JavaScript bundles — 5.091 s
11:41:56 AM: /usr/local/bin/build: line 32:  1737 Killed                  gatsby build
```

After working through this problem with Netlify's support who explained the app is being killed due to one of the plugins using vast amounts of memory, I've removed everything from my Gatsby project bit by bit until only this plugin is left. 

I've created a minimum example which you can find [here](https://github.com/Undistraction/undistraction-com/tree/gatsby-favicon-memory-error). As you will see there is literally nothing else in the project other than a layout, a simple page and this plugin. .

Just add the repo to Netlify and push. You will see the build fail. 

- gatsby-plugin-favicon 2.1.1
- gatsby 1.9.128

**Comment by kekami** [comment]
I just went through the same ordeal and can confirm that it was gatsby-plugin-favicon v2.1.1 that caused Netlify to kill the build process.

**Comment by Undistraction** [comment]
@kekami Yep. Ordeal is the right word. Lost a lot of time on this one. It was the first time I've used Netlify and was surprised how poor the debugging is. They said they had had problems with this plugin before, but looks like nobody had opened an issue. I've yet to try this in a V2 app, but I expect the same problem.

**Comment by julien1619** [comment]
Thanks for the report! I checked in the underlying plugin and it seems that it's loading ALL the thumbnails in RAM before writing it to the filesystem. I'll investigate on how to fix it.

In the meantime you can try one of the following:
- Enable `favicons` only, disable the others (`android`, `appleIcon`, ...).
- Disable the whole plugin and test `gatsby-plugin-manifest` (be warned, it will generate a PWA manifest, not only favicons).
- Don't use Netlify.
